django/core/management/commands/squashmigrations.py

```diff
@@ -75,7 +74,9 @@ def handle(self, **options):
             raise CommandError(str(err))
         # Load the current graph state, check the app and migration they asked
         # for exists.
-        loader = MigrationLoader(connections[DEFAULT_DB_ALIAS])
+        loader = MigrationLoader(
+            connections[DEFAULT_DB_ALIAS], replace_migrations=False
+        )
```

Comment:
I don't believe this is semantically what I wanted with the double squashing. If I squash A, B, and C into D, then D will have optimized migrations (I usually hand-optimize them too). Buy not using the operations from the squashed migration D, I'll lose all the work done.

This is the equivalent of having to re-resolve a merge conflict over and over again. Maybe fine in the end for most people? But not what I would be looking for as a result.

(Thank you for picking this up by the way! I had started working on this when I worked on a project with messy migrations but am no longer on that project...)

Reply:
I see your concern. I've changed it to first try using the squashed migrations. If that fails then it will recreate the migration graph with all squashed migrations removed (replaced by original migrations) and try again.